### PR TITLE
Add transaction trace utility function

### LINKED ISSUE #233: Feat request:  add `getTransactionTrace` to the `hre.starknet`
From discord:

It is useful in tests where I want to verify function return values. Right now I can 'force' the plugin to return the trace:

```
const hre: any = (starknet.devnet as any).hre;
const options = {
  feederGatewayUrl: adaptUrl(hre.config.starknet.networkUrl),
  gatewayUrl: adaptUrl(hre.config.starknet.networkUrl),
  hash: txHash
}

const preparedOptions = hre.starknetWrapper.prepareTxQueryOptions("get_transaction_trace", options);
const trace = await hre.starknetWrapper.execute("starknet", preparedOptions);
```

but it is not user friendly.

### PR DESCRIPTION
**Pull request by Nathan-SL**
## Usage related changes

-   Closes #233 
-   Get transaction trace using `const tx_trace = await starknet.getTransactionTrace(tx_hash);`

## Development related changes

-   Testing deployment with alpha-goerli2 config network is suppressed temporarily.

## Checklist:

-   [x] Formatted the code
-   [x] No linter errors + tried to avoid introducing linter warnings
-   [x] Performed a self-review of the code
-   [x] Rebased to the last commit of the target branch (or merged it into my branch)
-   [x] Documented the changes
    -   [ ] If `src/type-extensions.ts` was changed, updated the line number referencing this file in the `## API` section of README.md
-   [x] Updated the `test` directory (with a test case consisting of `network.json`, `hardhat.config.ts`, `check.sh`)
-   [x] Linked issues which this PR resolves
-   [x] Created a PR to the `plugin` branch of [`starknet-hardhat-example`](https://github.com/Shard-Labs/starknet-hardhat-example):https://github.com/Shard-Labs/starknet-hardhat-example/pull/85
    -   [x] Modified `test.sh` to use my example repo branch
    -   [x] Restored `test.sh` to to use the original branch (after the example repo PR has been merged)
-   [x] All tests are passing (for external contributors who don't have access to the CI/CD pipeline)

### CODE REVIEW

File: src/starknet-types.ts [hunk 1]
```diff
@@ -102,6 +102,27 @@ export interface Transaction {
...
+    events: unknown[];
+    execution_resources: ExecutionResources;
+    internal_calls: unknown[];
+    messages: unknown[];
+    result: string[];
+    selector: string;
+}
+
+export interface TransactionTrace {
+    function_invocation: FunctionInvocation;
```
Comment (FabijanC) [inline src/starknet-types.ts:121]: Some of these could be made optional (`?`) if they aren't present in all types of txs.

File: test/configuration-tests/with-networks/check.ts [hunk 2]
```diff
@@ -44,10 +44,10 @@ execution = hardhatStarknetTest("--no-compile test/contract-factory-test.ts".spl
...
 console.log("Success");
 
-console.log("Testing with alpha-goerli2 config network");
-process.env.NETWORK = "alpha-goerli2";
-hardhatStarknetDeploy(
-    "starknet-artifacts/contracts/contract.cairo --starknet-network alpha-goerli2 --inputs 10".split(
-        " "
-    )
-);
+// console.log("Testing with alpha-goerli2 config network");
```
Comment (FabijanC) [inline test/configuration-tests/with-networks/check.ts:47]: Suppression of this should be documented in development related changes in the PR description. Also, a message can be logged here about the suppression.

File: src/starknet-wrappers.ts [hunk 3]
```diff
@@ -265,6 +271,23 @@ export abstract class StarknetWrapper {
 
     public abstract getTxStatus(options: TxHashQueryWrapperOptions): Promise<ProcessResult>;
 
+    protected prepareTransactionTraceOptions(
```
Comment (FabijanC) [inline src/starknet-wrappers.ts:274]: Can prepareTxQueryOptions be used here?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: test/configuration-tests/with-networks/check.ts (hunk 2 above)
Comment (Nathan-SL) [reply test/configuration-tests/with-networks/check.ts:48]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/adc84b2a14c45a350409b4a31486ad57f2cfd08d

File: src/starknet-types.ts (hunk 1 above)
Comment (Nathan-SL) [reply src/starknet-types.ts:121]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/583dac4fa6885dabb6f19d31171947d5d9f70e6e

File: src/starknet-wrappers.ts (hunk 3 above)
Comment (Nathan-SL) [reply src/starknet-wrappers.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/9035f723400261e01d8eee31cc92f913c47f6e4b

Comment (FabijanC) [reply src/starknet-wrappers.ts]: I think even the argument type can be reused

Comment (FabijanC) [reply src/starknet-wrappers.ts]: I.e. we might not need the new options type

Comment (Nathan-SL) [reply src/starknet-wrappers.ts]: Great catch! https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/70a40a06ce8a1305e2fe460d0e78ecbf271e4a2e

File: src/starknet-types.ts [hunk 4]
```diff
@@ -102,6 +102,28 @@ export interface Transaction {
...
 }
 
+export interface FunctionInvocation {
+    call_type: string;
+    calldata: string[];
+    caller_address: string;
+    class_hash: string;
+    contract_address: string;
+    entry_point_type: string;
+    events: unknown[];
```
Comment (FabijanC) [inline src/starknet-types.ts:112]: How come it's using `unknown` here?

Comment (Nathan-SL) [reply src/starknet-types.ts]: https://github.com/Shard-Labs/starknet-hardhat-plugin/pull/264/commits/a38b37733e7611035080a544963379eb4af3ce43

**Pull request merged**